### PR TITLE
Issue 3015 - store timestamp when password is reset

### LINKED ISSUE #3015: Store time stamp when password reset is performed
This is simply a request to create a column some were (maybe in the users table) that when the user resets there password the date/time stamp of when the transaction happened is stored. 

Why is there not a request label?

### PR DESCRIPTION
**Pull request by kopcion**
## Related Ticket(s)
- Fixes #3015 

## Short roundup of the initial problem
Store timestamp when password is reset.

## What will change with this Pull Request?
- added new column to users table
- whenever password is changed timestamp is updated.

### CODE REVIEW

File: servatrice/src/serversocketinterface.cpp [hunk 1]
```diff
@@ -1794,3 +1792,9 @@ void WebsocketServerSocketInterface::binaryMessageReceived(const QByteArray &mes
 
     processCommandContainer(newCommandContainer);
 }
+
+// TODO make this configurable?
+bool isPasswordLongEnough(const int passwordLength) const
```
Comment (ctrlaltca) [inline servatrice/src/serversocketinterface.cpp]: Nice refactor, but this method implementation is missing the class name:
```
bool AbstractServerSocketInterface::isPasswordLongEnough(.....
```

Comment (kopcion) [reply servatrice/src/serversocketinterface.cpp]: Indeed, my bad. Fixed

File: servatrice/src/serversocketinterface.cpp [hunk 2]
```diff
@@ -1794,3 +1792,9 @@ void WebsocketServerSocketInterface::binaryMessageReceived(const QByteArray &mes
 
     processCommandContainer(newCommandContainer);
 }
+
+// TODO make this configurable?
+bool AbstractServerSocketInterface::isPasswordLongEnough(const int passwordLength)
+{
+    return passwordLength < 6;
```
Comment (ZeldaZach) [inline servatrice/src/serversocketinterface.cpp]: I'd like to see this configurable before merging, if possible <3

Comment (kopcion) [reply servatrice/src/serversocketinterface.cpp]: Sure thing, hopefully it will be done by Monday.

**Comment by skwerlman** [comment]
maybe im missing something, but what does this add over the existing `PASSWORD_RESET_REQUEST`, `PASSWORD_RESET_CHALLANGE` (which is typo'd in the code, im just noticing), and `PASSWORD_RESET` audit events, all of which are timestamped and include the relevent username?

**Comment by kopcion** [comment]
To be honest I didn't know that such thing has been implemented, as the issue is still open.

**Comment by tooomm** [comment]
Interesting note regarding this change: https://github.com/Cockatrice/Cockatrice/commit/cd431594e2fbbefe3ea0682e4f1adb4f3490719b#commitcomment-41185104

**Pull request merged**